compiler/hash-typecheck/src/traverse/mod.rs

```diff
@@ -391,11 +400,9 @@ impl<'tc> visitor::AstVisitor for TcVisitor<'tc> {
         } else {
             CORE_IDENTIFIERS.r#false
         });
-
-        // add the location of the term to the location storage
-        self.copy_location_from_node_to_target(node, term);
-
-        Ok(self.validator().validate_term(term)?.simplified_term_id)
+        let term = self.validator().validate_term(term)?.simplified_term_id;
+        self.register_node_info(node, term);
+        Ok(term)
```

Comment:
validation occurs before location is set on term... although that might not be relevant here

Reply:
Good catch!